src/details/ArborX_Polytope.hpp

```diff
@@ -0,0 +1,338 @@
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+#ifndef ARBORX_POLYTOPE_HPP
+#define ARBORX_POLYTOPE_HPP
+
+#include <ArborX_Box.hpp>
+#include <ArborX_DetailsAlgorithms.hpp>
+#include <ArborX_DetailsKokkosExt.hpp> // ArithmeticTraits
+#include <ArborX_DetailsTags.hpp>
+#include <ArborX_Point.hpp>
+#include <ArborX_Sphere.hpp>
+
+#include <Kokkos_Macros.hpp>
+
+namespace ArborX
+{
+
+namespace Details
+{
+struct Direction
+{
+  float _data[3];
+};
+
+template <int k>
+struct PolytopeDirections;
+
+template <>
+struct PolytopeDirections<6>
+{
+protected:
+  static constexpr int n_directions = 3;
+  static KOKKOS_FUNCTION Kokkos::Array<Direction, n_directions> directions()
+  {
+    static constexpr Kokkos::Array<Direction, n_directions> directions = {
+        Direction{1, 0, 0},
+        Direction{0, 1, 0},
+        Direction{0, 0, 1},
+    };
+    return directions;
+  }
+};
+
+template <>
+struct PolytopeDirections<18>
+{
+protected:
+  static constexpr int n_directions = 9;
+  static KOKKOS_FUNCTION Kokkos::Array<Direction, n_directions> directions()
+  {
+    static constexpr Kokkos::Array<Direction, n_directions> directions = {
+        Direction{1, 0, 0},
+        Direction{0, 1, 0},
+        Direction{0, 0, 1},
+        // edges
+        Direction{1, 1, 0},
+        Direction{1, 0, 1},
+        Direction{0, 1, 1},
+        Direction{1, -1, 0},
+        Direction{1, 0, -1},
+        Direction{0, 1, -1},
+    };
+    return directions;
+  }
+};
+
+template <>
+struct PolytopeDirections<26>
+{
+protected:
+  static constexpr int n_directions = 13;
+  static KOKKOS_FUNCTION Kokkos::Array<Direction, n_directions> directions()
+  {
+    static constexpr Kokkos::Array<Direction, n_directions> directions = {
+        Direction{1, 0, 0},
+        Direction{0, 1, 0},
+        Direction{0, 0, 1},
+        // edges
+        Direction{1, 1, 0},
+        Direction{1, 0, 1},
+        Direction{0, 1, 1},
+        Direction{1, -1, 0},
+        Direction{1, 0, -1},
+        Direction{0, 1, -1},
+        // corners
+        Direction{1, 1, 1},
+        Direction{1, -1, 1},
+        Direction{1, 1, -1},
+        Direction{1, -1, -1},
+    };
+    return directions;
+  }
+};
+KOKKOS_INLINE_FUNCTION float project(Point const &p, Direction const &d)
+{
+  float r = 0.;
+  for (int i = 0; i < 3; ++i)
+  {
+    r += p[i] * d._data[i];
+  }
+  return r;
+}
+} // namespace Details
+
+template <int k>
+struct kDOP : private Details::PolytopeDirections<k>
```

Comment:
What is the `sizeof(kDOP<6>)`? In the paper, they make a good point in that
```
since the k/2 defining directions are
fixed, the memory required to store each k-dop is only k
values (one value per plane), since the orientations of the
planes are known in advance.
```